### PR TITLE
[TEST] Split tests and skip file permission test on Windows

### PR DESCRIPTION
**Pull request by bizybot**
Changes to split tests for keytab file test cases instead of
randomized testing for testing branches in the code in the
same test.
On windows platform, for keytab file permission test, we 
required additional security permissions for the test 
framework. As this was the only test that required those
permissions, skipping that test on windows platform. 
The same scenario gets tested in *nix environments.

Closes#32768

### CODE REVIEW

**Comment by tvernum** [comment]
If we're changing permissions in the core test framework, then I'd like a review from someone in @elastic/es-core-infra

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/kerberos/KerberosRealmTests.java [hunk 1]
```diff
@@ -110,7 +110,6 @@ public void testLookupUser() {
         assertThat(future.actionGet(), is(nullValue()));
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/32768")
     public void testKerberosRealmWithInvalidKeytabPathConfigurations() throws IOException {
         final String keytabPathCase = randomFrom("keytabPathAsDirectory", "keytabFileDoesNotExist", "keytabPathWithNoReadPermissions");
```
Comment (jasontedor) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/kerberos/KerberosRealmTests.java]: I don't understand why we are using randomization here as opposed to simply running every case each time. Randomization is good for finding edge cases where we randomize some input data. I don't think it's good for poor man's test coverage to ensure all branches are covered, we really should be exercising those on every test run.

File: server/src/main/resources/org/elasticsearch/bootstrap/test-framework.policy [hunk 2]
```diff
@@ -40,6 +40,9 @@ grant codeBase "${codebase.lucene-test-framework}" {
   permission java.nio.file.LinkPermission "hard";
   // needed for RAMUsageTester
   permission java.lang.RuntimePermission "accessDeclaredMembers";
+  // needed for file permission tests to access user information
+  permission java.lang.RuntimePermission "accessUserInformation";
+  permission java.lang.RuntimePermission "getFileStoreAttributes";
```
Comment (jasontedor) [inline server/src/main/resources/org/elasticsearch/bootstrap/test-framework.policy]: If I understand correctly, we had something that was working on Linux and macOS but problematic on Windows? And the fix for Windows is now problematic on Linux and macOS? Can we merely use different logic on Windows versus Linux/macOS and avoid this problem altogether. If I am not understanding correctly and that is not possible, I would like us to avoid adding these permissions for a single test. Instead, could we get away with an evil test that does not use the security manager at all? This should be a last resort, I would like to know if the if (Windows) else (Linux/macOS) approach would be acceptable first.

**Review by jasontedor — CHANGES_REQUESTED** [review]
I left some comments.

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/kerberos/KerberosRealmTests.java [hunk 3]
```diff
@@ -110,52 +110,47 @@ public void testLookupUser() {
...
+            throw new UnsupportedOperationException(
+                    String.format(Locale.ROOT, "Don't know how to make file [%s] non-readable on a file system with attributes [%s]",
+                            dir.resolve(keytabFileName), supportedAttributes));
         }
+        final String expectedErrorMessage = "configured service key tab file [" + keytabPath + "] must have read permission";
+
+        assertKerberosRealmConstrutorFails(keytabPath.toString(), expectedErrorMessage);
+    }
 
+    private void assertKerberosRealmConstrutorFails(final String keytabPath, final String expectedErrorMessage) {
```
Comment (tvernum) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/kerberos/KerberosRealmTests.java]: s/`Construtor`/`Constructor`/g

**Review by tvernum — APPROVED** [review]

**Review by jasontedor — CHANGES_REQUESTED** [review]
Could you update the PR title and description, and ensure that when you commit this you commit with a summary and description that reflect the actual changes here?

**Review by jasontedor — APPROVED** [review]
LGTM.
